### PR TITLE
Hotfix  Brandon

### PR DESCRIPTION
**Pull request by with-shrey**

### CODE REVIEW

File: admin/src/app/home/orders/shared-components/category-anchor/category-anchor.component.html [hunk 1]
```diff
@@ -1,13 +1,19 @@
 <div *ngIf="categoryLabels.length > 0" class="row badge-row">
-  <div class="title">Go to Category:</div>
+  <div class="title">Go to {{ showBin ? 'Bin Location': 'Category' }}</div>
+  <div
+    (click)="labelSelected(undefined)"
```
Comment (kamal0808) [inline admin/src/app/home/orders/shared-components/category-anchor/category-anchor.component.html:4]: Should this be undefined here? Just being cautious.

File: admin/src/app/home/orders/stock-orders/generated/generated.component.ts [hunk 2]
```diff
@@ -500,6 +497,71 @@ export class GeneratedComponent implements OnInit, OnDestroy {
...
+    const index = list.findIndex((item) => item.id === lineItems.id);
+    if (index > -1) {
+      popedItem = list[index];
+      list.splice(index, 1)
+      if (list.length === 0 &&  count > 0){
+        this.reloadData();
+      }
+      success();
+    }
+    console.log(popedItem);
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/generated/generated.component.ts:511]: either remove this log or log that it's deleting item from list

File: admin/src/app/home/orders/stock-orders/generated/generated.component.ts [hunk 3]
```diff
@@ -500,6 +497,71 @@ export class GeneratedComponent implements OnInit, OnDestroy {
...
+    return popedItem;
+  }
+
+  addItemToList(lineItems, list, success) {
+    list.push(lineItems);
+    success();
+  }
+
+  moveItemToAnotherList(lineItems, data) {
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/generated/generated.component.ts:520]: check if this is an instant action on staging, or should you spend time in setting a loader for this. is there a possibility that the user clicks on button twice coz this function is slow?

File: admin/src/app/home/orders/stock-orders/generated/generated.component.ts [hunk 4]
```diff
@@ -518,8 +580,20 @@ export class GeneratedComponent implements OnInit, OnDestroy {
     }
     this.orgModelApi.updateAllStockOrderLineItemModels(this.userProfile.orgModelId, this.order.id, lineItemsIDs, data)
       .subscribe((res: any) => {
-          this.reloadData();
-        },
+        if (lineItems) {
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/generated/generated.component.ts:583]: what happens if the updateAPI fails? shouldn't the item move back to its original list with an error displayed?

**Review by kamal0808 — CHANGES_REQUESTED** [review]
added some comments

File: admin/src/app/home/orders/stock-orders/generated/generated.component.ts (hunk 3 above)
Comment (with-shrey) [reply admin/src/app/home/orders/stock-orders/generated/generated.component.ts:520]: The api loading stops after the item has moved

File: admin/src/app/home/orders/stock-orders/generated/generated.component.ts (hunk 4 above)
Comment (with-shrey) [reply admin/src/app/home/orders/stock-orders/generated/generated.component.ts:583]: If the Api fails teh item wont move to the new list

**Pull request merged**